src/BigQuery/BigQueryClient.php

```diff
@@ -0,0 +1,370 @@
+     *           to 3.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['projectId'])) {
+            throw new \InvalidArgumentException('A projectId is required.');
+        }
+
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::SCOPE];
+        }
+
+        $this->connection = new Rest($config);
+        $this->projectId = $config['projectId'];
+    }
+
+    /**
+     * Runs a BigQuery SQL query in a synchronous fashion. Rows are returned
+     * immediately as long as the query completes within a specified timeout. In
+     * the case that the query does not complete in the specified timeout, you
+     * are able to poll the query's status until it is complete.
+     *
+     * Example:
+     * ```
+     * $queryResults = $bigQuery->runQuery('SELECT * FROM [bigquery-public-data:usa_names.usa_1910_2013]');
+     *
+     * $isComplete = $queryResults->isComplete();
+     *
+     * while (!$isComplete) {
+     *     sleep(1); // let's wait for a moment...
+     *     $queryResults->reload(); // trigger a network request
+     *     $isComplete = $queryResults->isComplete(); // check the query's status
+     * }
+     *
+     * foreach ($queryResults->getRows() as $row) {
+     *     echo $row['name'];
+     * }
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/v2/jobs/query Query API documentation.
+     *
+     * @param string $query A BigQuery SQL query.
+     * @param array $options {
+     *     Configuration options.
+     *
+     *     @type int $maxResults The maximum number of rows to return per page
+     *           of results. Setting this flag to a small value such as 1000 and
+     *           then paging through results might improve reliability when the
+     *           query result set is large.
+     *     @type array $defaultDataset Specifies the default datasetId and
+     *           projectId to assume for any unqualified table names in the
+     *           query. If not set, all table names in the query string must be
+     *           qualified in the format 'datasetId.tableId'.
+     *     @type int $timeoutMs How long to wait for the query to complete, in
+     *           milliseconds. Defaults to 10000 milliseconds (10 seconds).
+     *     @type bool $useQueryCache Whether to look for the result in the query
+     *           cache.
+     *     @type bool $useLegacySql Specifies whether to use BigQuery's legacy
+     *           SQL dialect for this query.
+     * }
+     * @return QueryResults
+     */
+    public function runQuery($query, array $options = [])
+    {
+        $response = $this->connection->query([
+            'projectId' => $this->projectId,
+            'query' => $query
+        ] + $options);
+
+        return new QueryResults(
+            $this->connection,
+            $response['jobReference']['jobId'],
```

Comment:
Will `$response` ever not be an array, or not have a `jobReference` key? You might want to put some checks around this if so.